colossus/src/main/scala/colossus/service/ServiceServer.scala

```diff
@@ -89,11 +88,14 @@ extends Controller[I,O](codec, ControllerConfig(50)) {
   private val requestBuffer = collection.mutable.Queue[SyncPromise]()
   private var numRequests = 0
 
-  def idleCheck(period: Duration) {
+  override def idleCheck(period: Duration) {
+    super.idleCheck(period)
```

Comment:
Let's see if I can say this intelligently, or if this is even an intelligent comment:

  I've never been a fan of having code that get's structured in a way that if a function is overridden and the super isn't invoked..Bad Things Happen.  If the super's function absolutely needs to run(as in this case), I'm more a fan of making it so that you cannot override that function, but provide a hook which gets called at the end of the function.  That way children classes can override that function, add their own functionality and not accidentally fubar stuff.

Not saying we should change this..but def something I think we should strive to avoid if at all possible.